openedx_learning/apps/authoring/components/api.py

```diff
@@ -363,3 +387,171 @@ def create_component_version_content(
         learner_downloadable=learner_downloadable,
     )
     return cvrc
+
+
+class AssetError(StrEnum):
+    """Error codes related to fetching ComponentVersion assets."""
+    ASSET_PATH_NOT_FOUND_FOR_COMPONENT_VERSION = auto()
+    ASSET_NOT_LEARNER_DOWNLOADABLE = auto()
+    ASSET_HAS_NO_DOWNLOAD_FILE = auto()
+
+
+def _get_component_version_info_headers(component_version: ComponentVersion) -> dict[str, str]:
+    """
+    These are the headers we can derive based on a valid ComponentVersion.
+
+    These headers are intended to ease development and debugging, by showing
+    where this static asset is coming from. These headers will work even if
+    the asset path does not exist for this particular ComponentVersion.
+    """
+    component = component_version.component
+    learning_package = component.learning_package
+    return {
+        # Component
+        "X-Learning-Core-Component-Key": component.publishable_entity.key,
+        "X-Learning-Core-Component-Uuid": component.uuid,
+        # Component Version
+        "X-Learning-Core-Component-Version-Uuid": component_version.uuid,
+        "X-Learning-Core-Component-Version-Num": component_version.version_num,
+        # Learning Package
+        "X-Learning-Core-Learning-Package-Key": learning_package.key,
+        "X-Learning-Core-Learning-Package-Uuid": learning_package.uuid,
```

Comment:
Tiny tiny nit: I know these are mostly for debugging, but it seems weird to me to call these `X-Learning-Core-...` instead of something like `X-OEX-...`. Are these Open edX components that just happen to be implemented in the learning core package, or is Learning Core an end user brand?

Even from a developer perspective, the name "Learning Core" is not helpful for letting people know where the corresponding implementation is located, because the python package and github repo are both called `openedx-learning` which is not obviously the same thing as `Learning-Core`.

Reply:
I think something in middleware will force this to come out mixed case, i.e. `X-OEX-` will get turned into `X-Oex-`. Regardless, I'm fine with changing this. Maybe `X-Open-edX-Learning-Package-Key`?

Reply:
If there's no concern about length of the header name, then I'd say spell it out-- no room for ambiguity with `X-Open-edX-` :)